### PR TITLE
Fixing Linking on urls

### PR DESCRIPTION
**Pull request by vishalnarkhede**
Related ZD #9383

**Fixes**

- Ignore email address from generating og preview
- sanitize the urls before using Linking module from react-native

### CODE REVIEW

File: src/components/StatusUpdateForm.js [hunk 1]
```diff
@@ -29,8 +29,8 @@ const ImageState = Object.freeze({
   UPLOAD_FAILED: Symbol('upload_failed'),
 });
 
-const urlRegex = /(https?:\/\/[^\s]+)/gi;
-
+const urlRegex = /(?:[\w/:@-]+\.[\w/:@.-]*)+(?=\s|$)/g;
+const emailRegex = /.*@.*/ ;
```
Comment (vishalnarkhede) [inline src/components/StatusUpdateForm.js:33]: In discussion with Anatoly, for better regex!! Will replace it!

File: src/utils.js [hunk 2]
```diff
@@ -26,6 +26,15 @@ export function humanizeTimestamp(timestamp, tDateTimeParser) {
   return time.from(now);
 }
 
+// https://reactnative.dev/docs/linking
+export const sanitizeUrlForLinking = (url) => {
+  if (url.indexOf('http://') === -1 && url.indexOf('https://') === -1) {
```
Comment (virdesai) [inline src/utils.js:31]: pretty much the same thing
```suggestion
  if (url.indexOf('http') === -1) {
```

File: src/components/StatusUpdateForm.js (hunk 1 above)
Comment (virdesai) [reply src/components/StatusUpdateForm.js:33]: what if the url has an `@` in the query params for some reason?

File: src/utils.js (hunk 2 above)
Comment (virdesai) [reply src/utils.js:31]: alright updated it with regex

Comment (vishalnarkhede) [reply src/utils.js:31]: Thanks alot @virdesai :)

File: src/components/StatusUpdateForm.js [hunk 3]
```diff
@@ -29,8 +29,7 @@ const ImageState = Object.freeze({
   UPLOAD_FAILED: Symbol('upload_failed'),
 });
 
-const urlRegex = /(https?:\/\/[^\s]+)/gi;
-
+const urlRegex = /(?:\s|^)((?:https?:\/\/)?(?:[a-z0-9-]+(?:\.[a-z0-9-]+)+)(?::[0-9]+)?(?:\/(?:[^\s]+)?)?)/g;
```
Comment (vishalnarkhede) [inline src/components/StatusUpdateForm.js:32]: Thanks to @AnatolyRugalev !! Works pretty well - https://regex101.com/r/3ms9WF/1/

**Review by virdesai — APPROVED** [review]

**Pull request merged**